mongo/gridfs/bucket.go

```diff
@@ -455,6 +452,50 @@ func (b *Bucket) findChunks(ctx context.Context, fileID interface{}) (*mongo.Cur
 	return chunksCursor, nil
 }
 
+// returns true if the 2 index documents are equal
+func indexDocsEqual(expected, actual bsoncore.Document) (bool, error) {
+	if bytes.Equal(expected, actual) {
+		return true, nil
+	}
+
+	actualElems, err := actual.Elements()
+	if err != nil {
+		return false, err
+	}
+	expectedElems, err := expected.Elements()
+	if err != nil {
+		return false, err
+	}
+
+	if len(actualElems) != len(expectedElems) {
+		return false, nil
+	}
+
+	for idx, expectedElem := range expectedElems {
+		actualElem := actualElems[idx]
+		if actualElem.Key() != expectedElem.Key() {
+			return false, nil
+		}
+
+		actualVal := actualElem.Value()
+		expectedVal := expectedElem.Value()
+		actualInt, actualOK := actualVal.AsInt64OK()
+		expectedInt, expectedOK := expectedVal.AsInt64OK()
+
+		if !actualOK || !expectedOK {
+			if actualVal.Equal(expectedVal) {
+				continue
```

Comment:
Looks like this condition is untested.

Reply:
we should never hit this condition so i removed it

Reply:
Does this assume that no existing index on the gridfs collection has a non-numerical value type?  If so I don't think that's guaranteed.  For example, I think I've heard of users creating [hashed indexes](https://docs.mongodb.com/manual/indexes/#hashed-indexes) on GrifFS collections

Reply:
The indexes that the function that calls this is trying to create are only numerical, so if an index has a non-numerical value then we can assume they are not equal.

Reply:
I see.  Thanks.